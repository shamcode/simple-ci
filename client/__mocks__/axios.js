export default {
    get: jest.fn(),
    post: jest.fn(),
    create: jest.fn( function() {
        return this;
    } )
}