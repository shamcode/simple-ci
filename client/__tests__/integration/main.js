import setup, { app } from './helpers'
import axios from 'axios';
jest.mock( 'axios' );

beforeEach( () => {
    jest.resetModules();
    jest.clearAllMocks();
    setup();
} );

it( 'main page', async() => {
    expect.assertions( 3 );

    axios
        .use( 'get', '/projects', [
            { id: 1, name: 'Test', cwd: 'tmp' },
            { id: 2, name: 'Test2', cwd: 'home' }
        ] );

    await app.start( false );
    app.checkBody();
    await app.waitRendering();
    app.checkBody();
    expect( axios.mocks.get ).toHaveBeenCalledTimes( 1 );
} );
