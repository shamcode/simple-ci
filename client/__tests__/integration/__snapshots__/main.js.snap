// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main page 1`] = `
"<header>
  <a href=\\"\\" class=\\"logo\\">Simple CI</a>
</header>
<!--Header-->
<main>
  <div class=\\"page-data-loading\\">
    <div class=\\"sk-cube-grid\\">
      <div class=\\"sk-cube sk-cube1\\"></div>
      <div class=\\"sk-cube sk-cube2\\"></div>
      <div class=\\"sk-cube sk-cube3\\"></div>
      <div class=\\"sk-cube sk-cube4\\"></div>
      <div class=\\"sk-cube sk-cube5\\"></div>
      <div class=\\"sk-cube sk-cube6\\"></div>
      <div class=\\"sk-cube sk-cube7\\"></div>
      <div class=\\"sk-cube sk-cube8\\"></div>
      <div class=\\"sk-cube sk-cube9\\"></div>
    </div>
  </div>
  <!--PageDataLoading-->
  <!--if-->
</main>
<footer> Sham 2018 </footer>
<!--Footer-->"
`;

exports[`main page 2`] = `
"<header>
  <a href=\\"\\" class=\\"logo\\">Simple CI</a>
</header>
<!--Header-->
<main>
  <!--if-->
  <!--ErrorsPanel-->
  <div class=\\"project-list\\">
    <a class=\\"project-card\\" href=\\"#\\"><span class=\\"name\\">Test</span></a>
    <a class=\\"project-card\\" href=\\"#\\"><span class=\\"name\\">Test2</span></a>
    <!--for-->
    <a class=\\"project-card project-card-create\\" href=\\"/projects/create\\">
      <div class=\\"create-icon\\"></div>
    </a>
  </div>
  <!--ProjectsList-->
  <!--if-->
</main>
<footer> Sham 2018 </footer>
<!--Footer-->"
`;
